BUG: `read_sql_table` can't find views

IamGianluca opened the pull request:
closes #10750 

Add ability to 'read_sql_table' to read views and implement an unit test to check its behaviour

sinhrks committed:
ENH: duplicated and drop_duplicates now accept take=all kw

sinhrks committed:
TST: make assertion messages more understandable

sinhrks committed:
BUG: Categorical doesn't show tzinfo properly

Commit:
BUG: Fix bug which was preventing the inheritance of Series' names when only few of them exist, and new column names are not provided via the 'keys' argument. Closes #10698

Commit:
Add ability to 'read_sql_table' to read views and implemnt unit test to check behaviour

Commit:
Add ability to 'read_sql_table' to read views and implemnt unit test to check behaviour

Commit:
Update install.rst

- Added hint regarding pip install on low memory machines.
- Added hint to python 3 version of pandas from distributon repos.

jorisvandenbossche commented:
Looks good!

* Can you add a small note in the whatsnew file (in the 'other enhancements' section)
* You included the test in `_TestSQLApi` which only tests the API with sqlite. I think this is OK (I don't see potential differences between database flavours), but in that case the sql code for the different flavors is not needed.

Commit:
BUG: Allow 'read_sql_table' to read from views. Solves #10750.

Commit:
Merge remote-tracking branch 'origin/issue_10750_fix' into issue_10750_fix

Commit:
BUG: Allow 'read_sql_table' to read from views. Solves #10750.

Commit:
Merge remote-tracking branch 'origin/issue_10750_fix' into issue_10750_fix

Conflicts:
	pandas/io/tests/test_sql.py

Commit:
# This is a combination of 5 commits.
# The first commit's message is:
Merge

# This is the 2nd commit message:

BUG: Categorical doesn't show tzinfo properly

# This is the 3rd commit message:

ENH: duplicated and drop_duplicates now accept take=all kw

# This is the 4th commit message:

TST: make assertion messages more understandable

# This is the 5th commit message:

Update install.rst

- Added hint regarding pip install on low memory machines.
- Added hint to python 3 version of pandas from distributon repos.

Commit:
Add whatsnew note

Commit:
Add whatsnew note and remove redundant tests

Commit:
Merge remote-tracking branch 'upstream/master' into issue_10750_fix

Commit:
Merge remote-tracking branch 'origin/issue_10750_fix' into issue_10750_fix

jorisvandenbossche commented:
@IamGianluca for future reference, it is not needed to open a new PR if something went wrong. You can always rebase the current branch and force push again.
